src/grid/grid.component.ts

```diff
@@ -0,0 +1,630 @@
+                }
+
+    public ngOnInit(): void {
+        if (this.data) {
+            this.dataContainer = new DataContainer(this.data);
+            if (this.paging) {
+                this.dataContainer.state.paging = {
+                    index: 0,
+                    recordsPerPage: this.rowsPerPage,
+                };
+            }
+            // should we attach to all browsers and only when we have sorting??
+            if (this.hasSorting) {
+                this.dataContainer.state.sorting.strategy = new StableSortingStrategy();
+            }
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public ngAfterContentInit(): void {
+        if (this.data && this.autoGenerate) {
+            this.autogenerate();
+        } else {
+            this.checkColumns();
+        }
+        this.colDiffer = this.cols.changes.subscribe((_) => {
+            this.checkColumns();
+            this._detector.markForCheck();
+        });
+
+    }
+
+    /**
+     * @hidden
+     */
+    protected checkColumns(): void {
+        this.columns = this.cols.toArray();
+        this.columns.forEach((col, index) => {
+            col.index = index;
+            this.onColumnInit.emit({column: col});
+        });
+    }
+
+    /**
+     * @hidden
+     */
+    public ngDoCheck(): void {
+        let diff: IterableDiffer = this._differ.diff(this.data);
+        if (this.shouldDataBind || diff) {
+            this.onBeforeProcess.emit(this.dataContainer.state);
+            this.dataContainer.process();
+            this.shouldDataBind = false;
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public ngOnDestroy(): void {
+        this.colDiffer.unsubscribe();
+    }
+
+    public getColumnByIndex(index: number): any {
+        if (this.columns[index] !== undefined) {
+            return this.columns[index];
+        }
+    }
+
+    public getColumnByField(field: string): any {
+        return this.columns.find((col) => col.field === field);
+    }
+
+    /**
+     * Returns the cell at rowIndex/columnIndex.
+     *
+     * @param {number} rowIndex
+     * @param {number} columnIndex
+     * @returns
+     *
+     * @memberOf IgxGrid
+     */
+    public getCell(rowIndex: number, columnIndex: number): any {
+        let result: any = { dataItem: null, element: null };
+        let record: any = this.dataContainer.getRecordByIndex(rowIndex, DataAccess.TransformedData);
+        let element: any = this._elementRef.nativeElement.querySelector(`td[data-row="${rowIndex}"][data-col="${columnIndex}"]`);
+
+        if (record) {
+            result.dataItem = record[this.columns[columnIndex].field];
+        }
+
+        if (element) {
+            result.element = element;
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns
+     *
+     * @param {number} rowIndex
+     * @returns
+     *
+     * @memberOf IgxGrid
+     */
+    public getRow(rowIndex: number): any {
```

Comment:
Make a return type here. It should be a `GridRow` type, or something of the sort. The type should consist of record and element associated with the record, as well as collection of `GridCell` for this row.